src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift

```diff
@@ -66,54 +66,53 @@ final class CachedAppConfiguration {
 	private var promises = [(Result<CachedAppConfiguration.AppConfigResponse, Never>) -> Void]()
 	private var requestIsRunning: Bool { !promises.isEmpty }
 
+	private static let queue = DispatchQueue(label: "fetchAppConfiguration queue", qos: .userInitiated, attributes: .concurrent)
+
 	private func getAppConfig(with etag: String? = nil) -> Future<AppConfigResponse, Never> {
 		return Future { promise in
+			Self.queue.async(flags: .barrier) {
```

Comment:
Would it be easier to make the `DispatchQueue` serial? Then you wouldn't need add the `.barrier` flag. Just asking.

Reply:
To be honest, I'm just 'expanding' the scope of the queue to the whole `Future` initializer as a result of a spontaneous meeting 😅

The main issue we have is that the app configuration was fetched multiple times on app launch. If we find a way to fetch that config (or do the fallback to a cached/default one) we would not need such stunts. This would normally my approach to tackle such obscure issues rather than fixing symptoms. Unfortunately it's the last day before release and I'm somewhat forced to _hack_ this.

If you want to explore such a solution ('wait for config on app launch') I'm happy to provide help if needed.

Reply:
Fun fact. It's `sync` now.